jax/_src/lax/lax.py

```diff
@@ -122,6 +122,15 @@ def _try_broadcast_shapes(
         return None
   return tuple(result_shape)
 
+def asarray(x: ArrayLike) -> Array:
+  """Lightweight conversion of ArrayLike input to Array output."""
+  if isinstance(x, Array):
+    return x
+  elif isinstance(x, np.ndarray) or np.isscalar(x):
+    return api.device_put(x)
```

Comment:
Should this work inside a trace context?

Reply:
In a traced context `isinstanc(x, Array)` will return True if `x` itself is a tracer. If `x` is a container/pytree of tracers, then it should hit the TypeError on line 131